catalog/core/catalog-core-localstorageprovider/src/main/java/org/codice/ddf/catalog/content/impl/FileSystemStorageProvider.java

```diff
@@ -562,17 +582,17 @@ private ContentItem generateContentFile(
       Files.createDirectories(contentDirectory);
     }
 
+    long copySize;
     Path contentItemPath =
         Paths.get(contentDirectory.toAbsolutePath().toString(), item.getFilename());
     ByteSource byteSource;
 
-    long copy;
-
     if (storeReference != null) {
-      copy = item.getSize();
+      copySize = item.getSize();
+      String encryptedReference = crypter.encrypt(storeReference);
       Files.write(
           Paths.get(contentItemPath.toString() + "." + REF_EXT),
-          storeReference.getBytes(Charset.forName("UTF-8")));
+          encryptedReference.getBytes(Charset.defaultCharset()));
```

Comment:
✏️ I don't think we want to rely on the default charset here. This is going to be different depending on the OS, that is why we are using UTF-8 all over the place.